lib/money/rates_store/memory.rb

```diff
@@ -107,13 +133,20 @@ def each_rate(&block)
       #
       # @param [String] currency_iso_from The currency to exchange from.
       # @param [String] currency_iso_to The currency to exchange to.
+      # @param [Date] date Date of the exchange rate, ignored if not using a historical store.
       #
       # @return [String]
       #
       # @example
       #   rate_key_for("USD", "CAD") #=> "USD_TO_CAD"
-      def rate_key_for(currency_iso_from, currency_iso_to)
-        [currency_iso_from, currency_iso_to].join(INDEX_KEY_SEPARATOR).upcase
+      def rate_key_for(currency_iso_from, currency_iso_to, date = nil)
+        key = if @options[:historical]
+          key_convert = [currency_iso_from, currency_iso_to].join(INDEX_KEY_SEPARATOR)
+          [key_convert, date.to_s].join(INDEX_KEY_DATE_SEPARATOR) if date
```

Comment:
This is slightly problematic as `Date` is often used in place of `DateTime` and `Time`, which have a similar interface. I think it's worth being explicit about formatting data into a string

Reply:
Do you think the caller should do the string formatting, and `rate_key_for` should accept a string, or do you have other ideas?